test/new-e2e/utils/e2e/e2e.go

```diff
@@ -3,8 +3,218 @@
+//
+// Here is a simple example of E2E tests.
+// E2E tests use [testify Suite] and it is strongly recommended to read the documentation of
+// [testify Suite] if you are not familiar with it.
+//
+//	import (
+//		"testing"
+//
+//		"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+//	)
+//
+//	type vmSuite struct {
+//		e2e.Suite[e2e.VMEnv]
+//	}
+//
+//	func TestVMSuite(t *testing.T) {
+//		e2e.Run(t, &vmSuite{}, e2e.EC2VMStackDef())
+//	}
+//
+//	func (v *vmSuite) TestBasicVM() {
+//		v.Env().VM.Execute("ls")
+//	}
+//
+// Writing a E2E test requires several steps.
+//
+// 1. Define your own type with the embedded [e2e.Suite] struct.
+//
+//	type vmSuite struct {
+//		e2e.Suite[e2e.VMEnv]
+//	}
+//
+// [e2e.VMEnv] defines the components available in your environment. See "Using existing stack definition section" for more information.
+//
+// 2. Write a regular Go test function that runs the test suite using [e2e.Run].
+//
+//	func TestVMSuite(t *testing.T) {
+//		e2e.Run(t, &vmSuite{}, e2e.EC2VMStackDef())
+//	}
+//
+// The first argument of [e2e.Run] is an instance of type [*testing.T].
+//
+// The second argument is a pointer to the previous defined structure (&vmSuite{} in our example)
+//
+// The third parameter defines the environment. See "Using existing stack definition section" for more information about a stack definition.
+//
+// 3. Write your test function
+//
+//	func (v *vmSuite) TestBasicVM() {
+//		v.Env().VM.Execute("ls")
+//	}
+//
+// [suite.Env] gives the access to the components in your environment.
+//
+// Depending on your stack definition, [e2e.Suite.Env] can provide the following objects:
+//   - [client.VM]: A virtual machine where you can execute commands.
+//   - [client.Agent]: A struct that provides methods to run datadog agent commands.
+//
+// # Using an existing stack definition
+//
+// The stack definition defines the components available in your environment.
+//
+//	type vmSuite struct {
+//		e2e.Suite[e2e.VMEnv]
+//	}
+//
+//	func TestVMSuite(t *testing.T) {
+//		e2e.Run(t, &vmSuite{}, e2e.EC2VMStackDef())
+//	}
+//
+// In this example, the components available are defined by the struct [e2e.VMEnv] which contains a virtual machine.
+// The generic type of [e2e.Suite] must match the type of the stack definition.
+// In our example, [e2e.EC2VMStackDef] returns an instance of [*e2e.StackDefinition][[e2e.VMEnv]].
+//
+// The following default stack definitions are provided:
+//   - [e2e.EC2VMStackDef] creates an environment with a virtual machine. See [e2e.EC2VMStackDef] for more information about the supported options.
+//   - [e2e.AgentStackDef] creates an environment with an Agent installed on a virtual machine. See [e2e.AgentStackDef] for more information about the supported options.
+//
+// # Defining your stack definition
+//
+// In some special cases, you have to define a custom environment.
+// Here is an example on how to define an environment with Docker installed on a virtual machine.
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
```suggestion
// Here is an example of an environment with Docker installed on a virtual machine.
```